egs/callhome_egyptian/s5/local/callhome_prepare_dict.sh

```diff
@@ -55,7 +55,7 @@ $dir/extra_questions.txt || exit 1;
 
 # Add prons for laughter, noise, oov
 for w in `grep -v sil $dir/silence_phones.txt`; do
-sed -i "/\[$w\]/d" $tmpdir/lexicon.3
+grep -v -e "\[$w\]" $tmpdir/lexicon.3
```

Comment:
Here, we are deleting lines with phones like `[noise]` from the lexicon. This can be written as
```
perl -i -ne "print unless /\[$w\]/" $tmpdir/lexicon.3
```

Here again, as many passes are done over the file as there are phones to delete. You may want to concoct something that results in e. g. 
```
perl -i -ne "print unless /\[(laughter|noice|clap)\]/"  $tmpdir/lexicon.3
```
seen by perl, to process the file in a single pass. But that's, again, up to you.